code/modules/hydroponics/beekeeping/bee_smoker.dm

```diff
@@ -0,0 +1,118 @@
+/// multiplier to decide how much fuel we add to a smoker
+#define WEED_WINE_MULTIPLIER 0.2
+/// how much using it costs
+#define SINGLE_USE_COST 5
+
+/obj/item/bee_smoker
+	name = "bee smoker"
+	desc = "A device which can be used to hypnotize bees!"
+	icon = 'icons/obj/service/hydroponics/equipment.dmi'
+	icon_state = "bee_smoker"
+	inhand_icon_state = "bee_smoker"
+	lefthand_file = 'icons/mob/inhands/equipment/hydroponics_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/hydroponics_righthand.dmi'
+	item_flags = NOBLUDGEON
+	/// current level of fuel we have
+	var/current_herb_fuel = 50
+	/// maximum amount of fuel we can hold
+	var/max_herb_fuel = 50
+	/// are we currently activated?
+	var/activated = FALSE
+	/// sound to play when releasing smoke
+	var/datum/looping_sound/beesmoke/beesmoke_loop
+
+/obj/item/bee_smoker/Initialize(mapload)
+	. = ..()
+	beesmoke_loop = new(src)
+
+/obj/item/bee_smoker/attack_self(mob/user)
+	. = ..()
+	if(.)
+		return TRUE
+	if(!activated && current_herb_fuel <= 0)
+		user.balloon_alert(user, "no fuel!")
+		return TRUE
+	alter_state()
+	user.balloon_alert(user, "[activated ? "activated" : "deactivated"]")
+	return TRUE
+
+/obj/item/bee_smoker/afterattack(atom/attacked_atom, mob/living/user, proximity)
+	. = ..()
+
+	if(!proximity)
+		return
+
+	. |= AFTERATTACK_PROCESSED_ITEM
+
+	if(!activated)
+		user.balloon_alert(user, "not activated!")
+		return
+
+	if(current_herb_fuel < SINGLE_USE_COST)
+		user.balloon_alert(user, "not enough fuel!")
+		return
+
+	current_herb_fuel -= SINGLE_USE_COST
+	playsound(src, 'sound/effects/spray2.ogg', 100, TRUE)
+	var/turf/target_turf = get_turf(attacked_atom)
+	new /obj/effect/temp_visual/mook_dust(target_turf)
+
+	for(var/mob/living/basic/bee/friend in target_turf)
+		if(friend.flags_1 & HOLOGRAM_1)
+			continue
+		friend.befriend(user)
+
+	if(!istype(attacked_atom, /obj/structure/beebox))
+		return
+
+	var/obj/structure/beebox/hive = attacked_atom
+	for(var/mob/living/bee as anything in hive.bees)
+		if(bee.flags_1 & HOLOGRAM_1)
+			continue
+		bee.befriend(user)
+
+/obj/item/bee_smoker/attackby(obj/item/herb, mob/living/carbon/human/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+	if(!istype(herb, /obj/item/food/grown/cannabis))
+		return TRUE
```

Comment:
```suggestion
	if(!istype(herb, /obj/item/food/grown/cannabis))
		return 
```
Probably don't wanna say attackby was handled if we had the wrong type, though I don't think this'll matter presently.